pennylane/devices/default_qubit.py

```diff
@@ -586,9 +586,7 @@ def expval(self, observable, shot_range=None, bin_size=None):
             else:
                 # Coefficients and the state are not trainable, we can be more
                 # efficient in how we compute the Hamiltonian sparse matrix.
-                if observable.name == "Hamiltonian":
-                    Hmat = qml.utils.sparse_hamiltonian(observable, wires=self.wires)
-                elif observable.name == "SparseHamiltonian":
+                if observable.name in {"Hamiltonian", "SparseHamiltonian"}:
                     Hmat = observable.sparse_matrix()
```

Comment:
Sorry this is my mistake in what I suggested:
```suggestion
                    Hmat = observable.sparse_matrix(wires=self.wires)
```

This should fix some of the failing tests.